apps/settings/test/marionette/tests/keyboard_settings_test.js

```diff
@@ -35,4 +58,41 @@ marionette('manipulate keyboard settings', function() {
     // Check the icon is not shown as [X], which is for dialog style.
     assert.ok(!keyboardPanel.isDialog());
   });
+
+  test('Change keyboard language', function() {
+    // Keyboard location of special character Ñ
+    var specialKeySelector = 'button[data-keycode-upper="209"]';
+
+    // Tap 'Select Keyboards' button
+    keyboardPanel.tapSelectKeyboards();
+
+    // Select keyboard language, then click back to make it "stick"
+    keyboardPanel.selectEspanolKeyboard();
+    keyboardPanel.tapBackButton();
+    settingsApp.close();
+
+    // Launch the keyboard test app to verify the keyboard layout
+    keyboardTestApp.launch();
+    keyboardTestApp.textInput.tap();
+
+    // Switch to keyboard frame
+    systemInputMgmt = client.loader.getAppClass('system', 'input_management');
+    systemInputMgmt.waitForKeyboardFrameDisplayed();
+    systemInputMgmt.switchToActiveKeyboardFrame();
+
+    // Switch language
+    keyboardApp.imeSwitchingKey.tap();
+
+    // Try typing the special character
```

Comment:
The English layout contain this key too so the assertion in Gip is not really trusted.

I would recommend you assert the label of the displayed imeSwitchingKey before trying to locate (and tap) this special key.